tests/page_test.go

```diff
@@ -174,6 +174,84 @@ func TestPageEvaluate(t *testing.T) {
 	})
 }
 
+func TestPageEvaluateMapping(t *testing.T) {
+	tests := []struct {
+		name    string
+		script  string
+		want    any
+		wantErr string
+	}{
+		{
+			name:   "()_=>_0",
+			script: "() => 0",
+			want:   0,
+		},
+		{
+			name:   "function()_{return_1}",
+			script: "function() {return 1}",
+			want:   1,
+		},
+		{
+			name:   "()_=>_{2}",
+			script: "() => {2}",
+			want:   goja.Null(),
+		},
+		{
+			name:   "function()_{3}",
+			script: "function() {3}",
+			want:   goja.Null(),
+		},
+		{
+			name:   "async_function()_{return_4}",
+			script: "async function() {return 4}",
+			want:   4,
+		},
+		{
+			name:    "5",
+			script:  "5",
+			wantErr: "given expression does not evaluate to a function",
+		},
+		{
+			name:    "(6)",
+			script:  "(6)",
+			wantErr: "given expression does not evaluate to a function",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			_, rt, _, cleanUp := startIteration(t)
+			defer cleanUp()
+
+			// Test script as non string input
+			got, err := rt.RunString(fmt.Sprintf(`
+				const p = browser.newPage()
+				p.evaluate(%s)
+			`, tt.script))
+			if tt.wantErr != "" {
```

Comment:
I would write two top-level tests to reduce the conditional boilerplate in this test.

```go
func TestPageEvaluateMapping(t *testing.T) {
   ... happy path ...
}

func TestPageEvaluateMappingError(t *testing.T) {
   ... sad path ...
}
```

This way, we can make the test more readable. No strong opinion.

Reply:
I like this approach. This is how i prefer to split happy and error paths too 👍 I've amended it in 972a8da8ade8e5e9cf667cc5ee865ab2e7099f5e